lib/meta.nix

```diff
@@ -147,15 +146,36 @@ rec {
        => { shortName = "MY LICENSE"; }
   */
   getLicenseFromSpdxId =
-    let
-      spdxLicenses = lib.mapAttrs (id: ls: assert lib.length ls == 1; builtins.head ls)
-        (lib.groupBy (l: lib.toLower l.spdxId) (lib.filter (l: l ? spdxId) (lib.attrValues lib.licenses)));
-    in licstr:
-      spdxLicenses.${ lib.toLower licstr } or (
+    licstr:
+      getLicenseFromSpdxIdOr licstr (
         lib.warn "getLicenseFromSpdxId: No license matches the given SPDX ID: ${licstr}"
         { shortName = licstr; }
       );
 
+  /* Get the corresponding attribute in lib.licenses from the SPDX ID,
+     or fallback to the given default value.
+     For SPDX IDs, see https://spdx.org/licenses
+
+     Type:
+       getLicenseFromSpdxIdOr :: str -> Any -> Any
+
+     Example:
```

Comment:
Could you add some tests in `lib/tests/misc.nix`? These examples would work well :)

Reply:
Tests added for `getLicenseFromSpdxIdOr`